api/src/main/java/org/openmrs/api/AdministrationService.java

```diff
@@ -392,4 +392,71 @@ public interface AdministrationService extends OpenmrsService {
 	 * @since 2.4
 	 */
 	public void updatePostgresSequence();
+	
+	/**
+	 * Filters out all <code>globalProperties</code> to which given user does not have access. If user is not specified
+	 * then implementations should treat authenticated user from context as given user by default.<p>
+	 *
+	 * <strong>Should</strong> filter <code>globalProperties</code> if user is not allowed to see some <code>globalProperties</code>.<br>
+	 * <strong>Should</strong> not filter all <code>globalProperties</code> when the <code>globalProperty</code> view privilege column is null.
+	 *
+	 * @param properties the list of <code>globalProperties</code> to be filtered.
+	 * @param user the user instance to filter "visible" <code>globalProperties</code> for.
+	 * @return list, that does not include <code>globalProperties</code>, which can not be shown to given user due to
+	 *         permissions check.
+	 *
+	 * @since 2.7.0
+	 */
+	public List<GlobalProperty> filterGlobalPropertiesByViewPermissions(List<GlobalProperty> properties, User user);
```

Comment:
Because we are dealing with the authenticated user, and you are consistently passing null for the second parameter, what then do we need it for?

Reply:
Second question, do these methods need to be public? I thought that they are private internal implementation details that API consumers do not need to be aware of.

Reply:
I had made them public for testing

Reply:
Cant we just test the already public methods which use them internally?